R/geo-functions.R

```diff
@@ -176,13 +176,43 @@ bbox_scale <- function(bb, scale_factor){
 #' Takes a bounding box as an input and outputs a box in the form of a polygon
 #'
 #' @inheritParams gclip
+#' @aliases bb2poly
 #' @export
 #' @examples
-#' bb <- structure(c(-1.55080650299106, 53.8040984493515, -1.51186138683098,
-#' 53.828874094091), .Dim = c(2L, 2L), .Dimnames = list(c("coords.x1",
-#'   "coords.x2"), c("min", "max")))
-#' bb1 <- bb2poly(bb)
-#' plot(bb1)
+#' geo_bb(routes_fast)
+#' geo_bb(sp::bbox(routes_fast))
+#' # Simple features implementation:
+#' bb_sf1 <- geo_bb(routes_fast_sf) # keeps CRS
+#' plot(bb_sf1)
+#' plot(routes_fast, add = TRUE)
+#' geo_bb(sf::st_bbox(routes_fast_sf)) # no CRS in bbox
+geo_bb <- function(bb) {
+  UseMethod("geo_bb")
+}
+
+#' @export
+geo_bb.Spatial <- function(bb) {
+  bb2poly(bb)
+}
+
+#' @export
+geo_bb.sf <- function(bb) {
+  bb <- sf::as_Spatial(sf::st_geometry(bb))
```

Comment:
Yes but `as_Spatial()` should soon be exported by @edzer. See https://github.com/r-spatial/sf/commit/c2a493ed19482fdfc955c0bf9ab4a7eb133f1315 and my follow-on PR.